src/_P001_Switch.ino

```diff
@@ -173,51 +270,148 @@ boolean Plugin_001(byte function, struct EventStruct *event, String& string)
+              log += state ? F("1") : F("0");
+              log += F(" Output value ");
+              log += output_value;
+              addLog(LOG_LEVEL_INFO, log);
+              sendData(event);
             }
-            UserVar[event->BaseVarIndex] = output_value;
-            String log = F("SW   : Switch state ");
-            log += state ? F("1") : F("0");
-            log += F(" Output value ");
-            log += output_value;
-            addLog(LOG_LEVEL_INFO, log);
-            sendData(event);
+            Settings.TaskDevicePluginConfigLong[event->TaskIndex][0] = millis();
           }
         }
+        //check if LP is enabled and if LP has not fired yet
+        else if (Settings.TaskDevicePluginConfig[event->TaskIndex][5] && !Settings.TaskDevicePluginConfig[event->TaskIndex][6]) {
+          /**************************************************************************\
+          20181009 - @giig1967g: new longpress logic is:
+          if there is no 'state' change, check if longpress interval reached
+          When reached send longpress event.
+          Returned Event value = state + 10
+          So if state = 0 => EVENT longpress = 10
+          if state = 1 => EVENT longpress = 11
+          So we can trigger longpress for high or low contact
+
+          In rules this can be checked:
+          on Button#Switch=10 do //will fire if longpress when state = 0
+          on Button#Switch=11 do //will fire if longpress when state = 1
+          \**************************************************************************/
+          const unsigned long deltaLP = timePassedSince(Settings.TaskDevicePluginConfigLong[event->TaskIndex][2]);
+          if (deltaLP >= (unsigned long)lround(Settings.TaskDevicePluginConfigFloat[event->TaskIndex][2]))
+          {
+            byte output_value;
+            byte needToSendEvent = false;
+
+            Settings.TaskDevicePluginConfig[event->TaskIndex][6] = true;
+
+            switch(Settings.TaskDevicePluginConfig[event->TaskIndex][2])
+            {
+              case PLUGIN_001_BUTTON_TYPE_NORMAL_SWITCH:
+                  needToSendEvent = true;
+                break;
+              case PLUGIN_001_BUTTON_TYPE_PUSH_ACTIVE_LOW:
+                if (!state)
+                  needToSendEvent = true;
+                break;
+              case PLUGIN_001_BUTTON_TYPE_PUSH_ACTIVE_HIGH:
+                if (state)
+                  needToSendEvent = true;
+                break;
+            }
+
+            if (needToSendEvent) {
+              boolean sendState = state;
+              if (Settings.TaskDevicePin1Inversed[event->TaskIndex])
+                sendState = !sendState;
+              output_value = sendState ? 1 : 0;
+              output_value = output_value + 10;
+
+              UserVar[event->BaseVarIndex] = output_value;
+              String log = F("SW   : LongPress: Switch state ");
+              log += state ? F("1") : F("0");
```

Comment:
Just a remark.
Strings of a single character can be done as `'1'` instead of `F("1")`. A single character is faster and used less memory.
Short strings, less than 4 characters, use less memory when not using the `F()` macro and it is also faster.